android/src/main/kotlin/io/harness/flutter_cfsdk/FfFlutterClientSdkPlugin.kt

```diff
@@ -89,24 +89,31 @@ class FfFlutterClientSdkPlugin : FlutterPlugin, MethodCallHandler {
 
             val targetInstance = Target().identifier(target)
 
-            CfClient.getInstance().initialize(
 
-                application,
-                key,
-                conf,
-                targetInstance
+                CfClient.getInstance().initialize(
 
-            ) { auth, execResult ->
+                    application,
+                    key,
+                    conf,
+                    targetInstance
 
-                postToMainThread {
-                    print(auth.environment)
+                ) { auth, execResult ->
+                    if (execResult.error == null) {
 
-                    Handler(Looper.getMainLooper()).post {
+                        postToMainThread {
+                            print(auth.environment)
+
+                            Handler(Looper.getMainLooper()).post {
+
+                                result.success(execResult.isSuccess())
+                            }
 
-                        result.success(execResult.isSuccess())
+                        }
+                    }
+                    else {
+                        result.success(execResult.isSuccess)
```

Comment:
I thought about just returning `false` here - but that could mask any edge cases I haven't thought of, so decided to use the actual value which _should_ be false if we get here.